js/common/tests/shell/shell-document.js

```diff
@@ -2338,6 +2338,44 @@ function DatabaseDocumentSuiteReturnStuff () {
       assertTypeOf("string", res2._rev);
     },
 
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief use overwrite option
+////////////////////////////////////////////////////////////////////////////////
+
+    testInsertOverwrite : function () {
+      var docHandle = collection.insert({ a : 1});
+      var key = docHandle._key;
+
+      // normal insert with same key must fail!
+      try{
+        var res = collection.insert({a : 2, _key : key});
+        fail();
+      }
+      catch (err) {
+        assertEqual(ERRORS.ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED.code, err.errorNum);
+      }
+
+      // overwrite with same key must work
+      collection.insert({a : 2, _key: key},{overwrite:true});
```

Comment:
the value of "a" should actually be changed here, in order to see that the document was actually replaced!
Apart from that, we want to verify that the document got actually replaced, so it would be good to write *new* attributes here, and test that some *old* attributes actually vanished after carrying out the replace.